src/App.tsx

```diff
@@ -1,26 +1,506 @@
+  const [todosLength, setTodosLength] = useState<number>(0);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  function handleFilter(sort: string) {
+    switch (sort) {
+      case 'all':
+        setTodos(currentTodos => currentTodos);
+        break;
+      case 'active':
+        setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+        break;
+      case 'completed':
+        setTodos(currentTodos => currentTodos.filter(todo => todo.completed));
+        break;
+    }
+  }
+
+  function handleUpadateNewTitle(id: number, title: string) {
+    const todo = todos.find(currentTodo => currentTodo.id === id);
+
+    if (todo?.title === title) {
+      setItemEditingId(null);
+      setNewTitle('');
+
+      return;
+    }
+
+    if (title) {
+      setIsUpdating(id);
+      setErrorMessage(null);
+
+      if (!todo) {
+        return;
+      }
+
+      const updatedTitle = { title: title.trim() };
+
+      updateTodo(id, updatedTitle)
+        .then(() => {
+          setTodos(currentTodos =>
+            currentTodos.map(item =>
+              item.id === id ? { ...item, title: updatedTitle.title } : item,
+            ),
+          );
+
+          setAllTodos(currentTodos => {
+            return currentTodos.map(item =>
+              item.id === id ? { ...item, title: updatedTitle.title } : item,
+            );
+          });
+        })
+        .catch(() => {
+          setErrorMessage('Unable to update a todo');
+        })
+        .finally(() => {
+          setIsUpdating(null);
+        });
+    }
+
+    if (title.trim() === '') {
+      setDeleteTodoId(id);
+    }
+  }
+
+  function handleUpdateTodo(id: number) {
+    const todo = todos.find(currentTodo => currentTodo.id === id);
+
+    if (!todo) {
+      return;
+    }
+
+    const updatedStatus = { completed: !todo.completed };
+
+    if (updatedStatus) {
+      setIsUpdating(id);
+      setErrorMessage(null);
+
+      updateTodo(id, updatedStatus)
+        .then(() => {
+          setTodos(currentTodos =>
+            currentTodos.map(item =>
+              item.id === id
+                ? { ...item, completed: updatedStatus.completed }
+                : item,
+            ),
+          );
+
+          setAllTodos(currentTodos =>
+            currentTodos.map(item =>
+              item.id === id
+                ? { ...item, completed: updatedStatus.completed }
+                : item,
+            ),
+          );
+
+          handleFilter(sortBy);
+
+          if (updatedStatus.completed) {
+            setItemsLeft(currentItemsLeft => currentItemsLeft - 1);
+          } else {
+            setItemsLeft(currentItemsLeft => currentItemsLeft + 1);
+          }
+        })
+        .catch(() => {
+          setErrorMessage('Unable to update a todo');
+        })
+        .finally(() => {
+          setIsUpdating(null);
+        });
+    }
   }
 
+  function handleUpdateAllTodos() {
```

Comment:
Only the last todo has loader when I click toggle all:
![image](https://github.com/user-attachments/assets/26bd23cc-cfbb-4da1-98fa-29edd3e47e62)